clang/lib/Driver/Driver.cpp

```diff
@@ -776,20 +767,20 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
           FoundNormalizedTriples[NormalizedName] = Val;
 
           // If the specified target is invalid, emit a diagnostic.
-          if (TT.getArch() == llvm::Triple::UnknownArch)
+          if (TT.getArch() == llvm::Triple::UnknownArch ||
+              !(TT.getArch() == llvm::Triple::spir ||
```

Comment:
@agozillon, could you extended driver tests to cover this check, please?

Reply:
Can do!

Reply:
I actually think this is covered already inside sycl-offload.c: https://github.com/intel/llvm/blob/sycl/clang/test/Driver/sycl-offload.c#L10 checks if it's invalid 

I could add a test to check if fsycl-targets fails when a real but not unlisted fsycl-target is specified like x86_64 though?

Reply:
Yes. I think the triple with valid architecture should cover new condition. Thanks!

Reply:
No problem, I added the new test case.